### PR TITLE
Add functionality for replacing more crops

### PR DESCRIPTION
**Pull request by temp1011**
Same as https://github.com/MinecraftForge/MinecraftForge/pull/4744 but with a branch that works.

### CODE REVIEW

**Comment by temp1011** [comment]
I still think this a useful feature for forge.

**Comment by mezz** [comment]
Wont this cause issues with existing `EnumPlantType.CROP` vanilla plants?
What do villagers do when they encounter a melon or pumpkin stem?

File: patches/minecraft/net/minecraft/entity/ai/EntityAIHarvestFarmland.java.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIHarvestFarmland.java
++++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIHarvestFarmland.java
+@@ -94,6 +94,8 @@
+                             world.func_180501_a(blockpos, Blocks.field_185773_cZ.func_176223_P(), 3);
+                             flag = true;
+                         }
++                        else if (itemstack.func_77973_b() instanceof net.minecraftforge.common.IPlantable && ((net.minecraftforge.common.IPlantable)itemstack.func_77973_b()).getPlantType(world, blockpos)==net.minecraftforge.common.EnumPlantType.Crop) {
++                        	world.func_180501_a(blockpos, ((net.minecraftforge.common.IPlantable)itemstack.func_77973_b()).getPlant(world, blockpos), 3); flag = true; }
```
Comment (mezz) [inline patches/minecraft/net/minecraft/entity/ai/EntityAIHarvestFarmland.java.patch]: This should be split across more lines, putting multiple statements together on one line and having the closing brace at the end of a statement makes this section hard to understand.

**Review by mezz — CHANGES_REQUESTED** [review]

**Comment by temp1011** [comment]
The villagers already automatically harvest anything that is an instance of `BlockCrops` and is fully grown. This PR just makes sure they also have the ability to replant the seeds of the crops that they do break. So I don't think melons and pumpkins will cause any problems since villagers don't break them anyway.

I've made the patch more readable. Hopefully it's not too large now.

**Comment by mezz** [comment]
Tested and looks good to me.

**Review by mezz — APPROVED** [review]

**Comment by Snownee** [comment]
@temp1011 Since villager naturally can only pick up vanilla item, so are these codes really reachable?

**Comment by temp1011** [comment]
This does seem problematic. I'll look into an updated and improved version of this once the planting interfaces are resolved.

**Pull request merged**